### PR TITLE
Masked select entry for pytorch

### LINKED ISSUE #5630: [Term Entry] PyTorch Tensor Operations .masked_select()
### Reason/inspiration (optional)

We would like a new entry on the `.masked_select()` term under PyTorch. The entry should go in a new file under `docs/content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept
- An `Example` section that provides an example demonstrating the concept in use

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by arisdelacruz**
### Description

This is an attempt to create a masked select entry for pytorch

### Issue Solved

Closes #5630 

### Type of Change

- Adding a new entry

### Checklist

- [/] All writings are my own.
- [/] My entry follows the Codecademy Docs style guide.
- [/] My changes generate no new warnings.
- [/] I have performed a self-review of my own writing and code.
- [/] I have checked my entry and corrected any misspellings.
- [/] I have made corresponding changes to the documentation if needed.
- [/] I have confirmed my changes are not being pushed from my forked `main` branch.
- [/] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [/] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+Title: 'masked_select()'
+Description: 'Selects elements from a tensor based on a mask.'
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md:3]:
```suggestion
Description: 'Selects elements from a tensor, based on a boolean mask, and returns them as a 1D tensor.'
```

File: content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  - 'Data Structures'
+  - 'Functions'
+  - 'Index'
+  - 'Values'
+CatalogContent:
+  - 'intro-to-py-torch-and-neural-networks'
+  - 'paths/computer-science'
+---
+
+In PyTorch, **masked select** is a method that selects elements from a tensor based on a mask. The mask is a tensor of the same shape as the input tensor, containing boolean values. The `masked_select()` function returns a new tensor with elements from the input tensor where the corresponding mask value is `True`.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md:17]:
```suggestion
In PyTorch, **`.masked_select()`** is a function that selects elements from an input tensor based on a boolean mask of the same shape. It returns a new 1D tensor containing the elements where the mask is `True`.
```

File: content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+## Syntax
+
+```pseudo
+torch.masked_select(input, mask, *, out=None) → Tensor
+```
+
+- `input`: The input tensor from which elements will be selected.
+- `mask`: The mask tensor containing boolean values to determine which elements to select.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md:26]:
```suggestion
- `mask`: A boolean tensor of the same shape as input, where `True` indicates the elements to be selected.
```

File: content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+- `out`: If provided, the result will be written to this tensor. It is an optional parameter.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md:27]:
```suggestion
- `out` (Optional): A tensor to store the result. If provided, the selected elements will be written to this tensor instead of creating a new one.
```

File: content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+Here's an example of using `masked_select()` in PyTorch:
+
+```py
+import torch
+
+# Create an input tensor
+input_tensor = torch.tensor([1, 2, 3, 4, 5])
+
+# Create a mask tensor
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md:39]:
```suggestion
# Create a mask tensor with boolean values
```

File: content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+mask = torch.tensor([True, False, True, False, True])
+
+# Use masked_select to select elements based on the mask
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md:42]:
```suggestion
# Use masked_select to extract elements from the input tensor where the mask is True
```

File: content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+Output:
+
+```shell
+tensor([1, 3, 5])
+```
+
+In this example, the `input_tensor` contains elements `[1, 2, 3, 4, 5]`, and the `mask` tensor contains boolean values `[True, False, True, False, True]`. The `masked_select()` function selects elements from the `input_tensor` where the corresponding mask value is `True`, resulting in the tensor `[1, 3, 5]`.
+
+`masked_select()` is useful for filtering elements from a tensor based on a condition specified by the mask tensor. It can be used in various scenarios, such as selecting specific elements for further processing or analysis.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md:57]:
```suggestion
The `.masked_select()` function is useful for filtering elements from a tensor based on conditions specified by the mask tensor. It can be applied in various scenarios, such as selecting specific elements for further processing, analysis, or model training."
```

File: content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
+---
+Title: 'masked_select()'
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md:2]:
```suggestion
Title: '.masked_select()'
```

File: content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+torch.masked_select(input, mask, *, out=None) → Tensor
+```
+
+- `input`: The input tensor from which elements will be selected.
+- `mask`: The mask tensor containing boolean values to determine which elements to select.
+- `out`: If provided, the result will be written to this tensor. It is an optional parameter.
+
+## Example
+
+Here's an example of using `masked_select()` in PyTorch:
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md:31]:
```suggestion
Here's an example of using `.masked_select()` in PyTorch:
```

File: content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+mask = torch.tensor([True, False, True, False, True])
+
+# Use masked_select to select elements based on the mask
+selected_elements = torch.masked_select(input_tensor, mask)
+
+# Print the selected elements
+print(selected_elements)
+```
+
+Output:
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/masked-select/masked-select.md:49]:
```suggestion
The code above generates the output as follows:
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @arisdelacruz thank you for contributing to Codecademy Docs, the entry is nicely written! 😄 

I've suggested a few changes, could you please review and modify those at your earliest convenience? Thank you! 😃

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @arisdelacruz  😄 

The entry looks good for a second round of review! 🚀

**Review by PragatiVerma18 — APPROVED** [review]
Hey @arisdelacruz, thanks for contributing this entry. This looks good to be merged! 🚀
